### PR TITLE
[Relay][VM] Fix code generation for packed functions + tuples 

### PR DESCRIPTION
**Pull request by jroesch**
Disabling fusion for concat in #3268 uncovered a bug in the VM compiler, this fixes it.

We were previously not unpacking tuples, this changes code generation to unpack the tuples into registers before invoking the function. 

cc @wweic @zhiics @icemelon9 @MarisaKirisame

### CODE REVIEW

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -292,7 +292,7 @@ RELAY_REGISTER_OP("concatenate")
 .add_type_rel("Concatenate", ConcatenateRel)
 .set_attr<FInferCorrectLayout>("FInferCorrectLayout", ConcatenateLayout)
 .set_attr<FTVMCompute>("FTVMCompute", ConcatenateCompute)
-.set_attr<TOpPattern>("TOpPattern", kInjective);
+.set_attr<TOpPattern>("TOpPattern", kOpaque);
```
Comment (wweic) [inline src/relay/op/tensor/transform.cc]: Maybe leave this line out of the PR?

**Comment by tqchen** [comment]
Please remove concat opaque property for now, and consider manually construct a function that needs to deal with such a case

**Comment by jroesch** [comment]
I don't think it is possible to write such a function because anything that uses tuples gets fused currently.

File: src/relay/backend/vm/compiler.cc [hunk 2]
```diff
@@ -334,15 +334,42 @@ struct VMCompiler : ExprFunctor<void(const Expr& expr)> {
...
+    CHECK_EQ(func->params.size(), args_registers.size());
+    for (size_t i = 0; i < func->params.size(); i++) {
+      auto ty = func->params[i]->checked_type();
+      if (ty.as<TensorTypeNode>()) {
+        unpacked_arg_regs.push_back(args_registers[i]);
+        arity += 1;
+      } else if (auto tuple_ty = ty.as<TupleTypeNode>()) {
+        for (size_t f = 0; f < tuple_ty->fields.size(); f++) {
+          const auto& field = tuple_ty->fields[f];
+          CHECK(field.as<TensorTypeNode>())
```
Comment (MarisaKirisame) [inline src/relay/backend/vm/compiler.cc]: can you extend it and allow recursive tuple? it is more uniform this way.

Comment (jroesch) [reply src/relay/backend/vm/compiler.cc]: I don't really want to do it right now, recursive is a pain given the way the code generator is written right now. This is mostly to fix CI breakage, I would like to make some updates to compiler likely after FCRC tutorial.

**Comment by tqchen** [comment]
Thanks, @jroesch @MarisaKirisame @wweic , this PR is now merged

**Pull request merged**